### ISSUE TITLE
Added update-type and remove-type

### ISSUE THREAD

**Issue by tobkle** [tmeasday/create-graphql-server#29]
@tmeasday Thanks for your great work again. This helped me a lot to get deeper into graphql.

Forked and enhanced your version for having beside add-type also ...

* update-type
* remove-type 

You will find it in my fork: https://github.com/tobkle/create-graphql-server

Be aware, that it doesn't update adjusted type specific resolvers, schemas or model files. Instead it will remove the existing files and will regenerate them. But during removal: Instead of deleting the files, it will archive them in specific archive folders. It removes the references of the types in the corresponding index.js files. See Changelog and adjusted documentation for more information.

Please be careful, this change is still an alpha version, and hasn't been tested well in my enhancements. Please feel free to copy or adjust to meet your own coding quality standards.

**Comment by tmeasday** [comment]
Hi @tobkle! Thanks for your contribution, this is a sorely needed feature (see #12).

I do wonder if archiving is really necessary though. Isn't it enough to assume users will use some kind of version control (i.e. git) and older versions of files will be available there?

It would be great to figure out a testing strategy for this as well. Perhaps we need two versions of the output app? Any thoughts there?

**Comment by tobkle** [comment]
You are right. Removed this archiving logic with the latest commit by now. 
Did also some code refactoring and commenting, (not yet JSDOC or something like). 
Fixed also the issue with your commented XXX to check for already existing references in the index files.

Testing logic is still open. 
I suppose, ...
-it should check for empty type input
-it should check, if files are generated, updated, deleted with the correct input (generated)
-it should check, if references are added only once
-it should check, if there is afterwards the corresponding <type> file for an existing reference
...?

Why do you think we need two versions of the output app? There I can't follow your thoughts.

What I was wondering is, what happens in an "update-type" run, if the user had changed a generated file. Do we just overwrite it, or do we provide an option "-force-update" or "-warn-on-changes". Does it make sense to calculate a checksum for a generated file and keep it somewhere, to figure out, if a file was updated or if it is still in its initial state? Or can we separate generated and adjusted code? Maybe an hooking mechanism like in feather.js? 

Concerning the User type and the Authentication mechanism you've explained in the Readme: Wouldn't it make sense, to have the User type already implemented with some kind of an authorization/role mechanism (e.g. in the Meteor style)? Or as an optional package? For many (also for me) this is usually one of the hardest parts, to get this done correctly and save.

**Comment by tmeasday** [comment]
> Why do you think we need two versions of the output app? There I can't follow your thoughts.

I guess I meant for this test: https://github.com/tmeasday/create-graphql-server/blob/master/test/output-app-generation-test.sh

We would need more copies of the output app if we were to add a line that updates or removes a type.

It would probably be good to have something though, just to ensure we maintain the correct behavior (especially testing around the correct includes in `index.js`).

My idea would be to run the first few `add-type`s, check against one output app, then run an `update-type`, check against another app, then `remove-type` and check against a final app.

> Do we just overwrite it, or do we provide an option "-force-update" or "-warn-on-changes".

I'd love to do this if we (as you've hinted at) can figure out a good method of detecting changes.

> Does it make sense to calculate a checksum for a generated file and keep it somewhere, to figure out, if a file was updated or if it is still in its initial state?

I'm interested in this idea. Where do you think we should keep it? Some ideas:

 1. A `.generated-checksums` file or the like
 2. A comment at the top of each file
 3. A corresponding `X.graphql.checksum` file

I think I prefer 1., but I'd like to hear your thoughts.

In general I would steer clear of trying to update the files in place however (i.e. you would necessarily lose your changes).

> Or can we separate generated and adjusted code? Maybe an hooking mechanism like in feather.js?

I think this runs counter to the philosophy of CGS, where the idea is that the generated code is easy to understand and highly modifiable.

> Wouldn't it make sense, to have the User type already implemented with some kind of an authorization/role mechanism (e.g. in the Meteor style)?

I added the user stuff late in the piece and I think it could be improved. Let's open a separate issue to discuss this; let me know your ideas.

**Comment by tmeasday** [comment]
PS you should open a PR and we should discuss the changes there.
